kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/CredentialsValidator.java

```diff
@@ -54,7 +54,7 @@ public CredentialsValidator() {
   @Override
   protected Optional<String> doValidate(BigQuerySinkConfig config) {
     String keyFile = config.getKey();
-    if (keyFile == null || keyFile.isEmpty()) {
+    if (keyFile == null || keyFile.isEmpty() || config.getKeySource() == GcpClientBuilder.KeySource.SERVICE_ACCOUNT) {
```

Comment:
It looks like we're skipping preflight validation when this new key source is used. Is there a reason for that? It seems like it'd still be useful to verify that we can construct a client in this case, especially since the docs for [`GoogleCredentials::getApplicationDefault`](https://googleapis.dev/java/google-auth-library/0.21.0/com/google/auth/oauth2/GoogleCredentials.html#getApplicationDefault--) indicate that it may throw an `IOException` if the credentials cannot be created.

Reply:
Also, I wonder if we may want to throw an error if the user has configured the connector with a non-empty `keyfile` and a `keySource` of `SERVICE_ACCOUNT` letting them know that they should either change `keySource` to `JSON` or `FILE`, or not provide a value for the `keyfile` property, since the keyfile is ignored when the keysource is `SERVICE_ACCOUNT` .

Reply:
Yes, I've restored the preflight validation and also added a new check to enforce that the `keyfile` must not be provided if `keySource=APPLICATION_DEFAULT`. Also covered by an unit test.